[Qin Nanxin] iP

InfiBeyond opened the pull request:
## Duke Max

> “Hi, I am Duke Max. Your personal task management companion."

Duke Max[ frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

InfiBeyond committed:
[Level 0. Rename, Greet, Exit]
(a) Changed chatbot name form Duke to Edna
(b) Implement an initial skeletal version

InfiBeyond committed:
[Level 1. Echo]
Improve the skeletal version of Edna so that it echos commands entered by the user, and exits when the user types the command bye.

InfiBeyond committed:
[Level 2. Add, List]
Add the ability to store whatever text entered by the user and display them back to the user when requested.

InfiBeyond committed:
[Level 3. Mark as Done]
(a) Create Task class.
(b) Add the ability to mark tasks as done.
(c) Add the ability to change the status back to not done.

InfiBeyond committed:
[Level 4. ToDos, Events, Deadlines]
Add support for tracking three types of tasks:
ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

* Fixed Level 3 error for mixing up "X" and " " for marked and unmarked tasks.

InfiBeyond committed:
[Automated Text UI Testing]
Use the input/output redirection technique to semi-automate the testing of Duke.

InfiBeyond committed:
[Level 5. Handle Errors]
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

Use Exceptions to handle errors
Define a class DukeException to represent exceptions specific to Duke.

InfiBeyond committed:
[Level 6. Delete]
Add support for deleting tasks from the list.

InfiBeyond committed:
Add DukeException class
Previously missed this file during git push.

InfiBeyond committed:
Previous files corrupted -> Have to fork project again [Level-0]
- Unable figure out why my computer cannot open the previous files
- Attempt to clone work from GitHub  didn't work

Level 0. Rename, Greet, Exit
(a) Rename chatbot to "Edna-Duke".
(b) Implement an initial skeletal version of the it that simply greets the user and exits.

InfiBeyond committed:
[Level 1. Echo]
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye.

InfiBeyond committed:
[Level 2. Add, List]
Add the ability to store whatever text entered by the user and display them back to the user when requested.

InfiBeyond committed:
[Level 3. Mark as Done]
Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

InfiBeyond committed:
[Level 4. ToDos, Events, Deadlines]
Add support for tracking three types of tasks:
- ToDos: tasks without any date/time attached to it e.g., visit new theme park
- Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
- Events: tasks that start at a specific date/time and ends at a specific date/time
  e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

InfiBeyond committed:
[Level 5. Handle Errors]
Teach the chatbot to deal with errors such as incorrect inputs entered by the user.

InfiBeyond committed:
[Level 6. Delete]
Add support for deleting tasks from the list.

mounilsankar commented on src/main/java/Deadline.java:
Overall good job!  Good use of abstraction using the Task class! However, it might be a better idea to have even more abstraction through the addition of a UI class that handles all the printing and a TaskList class that handles the list of Tasks.

mounilsankar commented on src/main/java/Duke.java:
The indentation for the case blocks inside the switch statement within the try block does not follow the coding standard

AlainS87 commented on src/main/java/Duke.java:
Indentation & Formatting:

The code seems to be properly indented, which makes it easier to read.
The consistent use of brackets {} and their placement adheres to the Java coding conventions.
Variable Naming:

Names like input, eventCommand, and ddlCommand are meaningful, but some variables like ddl could have a more descriptive name.
Variables should be named in a manner that makes their purpose clear. For instance, ddl might be better named as deadlineDetails or similar.
sc has not been defined in the given snippet, so it's unclear what it represents. If it's a Scanner object, a more intuitive name like scanner might be more appropriate.

AlainS87 commented on src/main/java/ToDo.java:
This program has only finished Week1-2 tasks, more OOP should be added

AlainS87 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,94 @@
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(partition);
     }
 }
```
Indentation & Formatting:

The code seems to be properly indented, which makes it easier to read.
The consistent use of brackets {} and their placement adheres to the Java coding conventions.
Variable Naming:

Names like input, eventCommand, and ddlCommand are meaningful, but some variables like ddl could have a more descriptive name.
Variables should be named in a manner that makes their purpose clear. For instance, ddl might be better named as deadlineDetails or similar.
sc has not been defined in the given snippet, so it's unclear what it represents. If it's a Scanner object, a more intuitive name like scanner might be more appropriate.

AlainS87 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,13 @@
+    public String getStatus() {
+        return "[To-Do]" + super.getStatus();
+    }
+}
```
This program has only finished Week1-2 tasks, more OOP should be added

andrefoo commented on src/main/java/Duke.java:
```diff
@@ -26,6 +26,10 @@ public static void main(String[] args) {
                         int unmarkItem = sc.nextInt();
                         taskList.get(unmarkItem - 1).markUndone();
                         break;
+                    case "delete":
```
Consider making inputs case-insensitive! 

This way, if I requested to add a todo writing: tOdO it would still add the task into the list.
Try using 'String' 's equalsIgnoreCase() method compares two strings, ignoring lower case and upper case differences.

andrefoo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,94 @@
+import java.util.*;
+
 public class Duke {
+    public static String partition = "------------------------------------------------------------";
```
Consider changing 'partition' to a final variable as it does not change, as well as fully capitalising the variable name to be 'PARTITION'

andrefoo reviewed (commented):
LGTM, some minor amendments needed that's all.

InfiBeyond committed:
[Level 7. Save]
Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.

InfiBeyond committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  [Level 7. Save] Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.

InfiBeyond committed:
[Level 8. Dates and Times
Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

InfiBeyond committed:
[Level 8. Dates and Times]
Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

InfiBeyond committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  [Level 8. Dates and Times Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

InfiBeyond committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  [Level 8. Dates and Times] Teach the chatbot how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, the chatbot should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

Kevin-Liusx commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,291 @@
+            } 
+
+            catch (DukeException e) {
+                System.out.println(e.getMessage());
```
I think your method is too long here. Maybe try to create some other methods?😎

InfiBeyond committed:
OOP part 1
change laptop

InfiBeyond committed:
[A-MoreOOP]
↳ Make the code more OOP

Refactor the code to extract out closely related code as classes.

InfiBeyond committed:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  [A-MoreOOP] ↳ Make the code more OOP
  OOP part 1 change laptop

InfiBeyond committed:
[A-Packages]
↳ Divide classes into packages

Organize the classes into suitable java packages.

InfiBeyond committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  [A-Packages] ↳ Divide classes into packages

InfiBeyond committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support

InfiBeyond committed:
Fix renaming import package error

InfiBeyond committed:
[A-Gradle]
↳ Automate project builds using Gradle

Use Gradle to automate some of the build tasks of the project.

InfiBeyond committed:
[A-JUnit]
↳ Add JUnit tests

Add JUnit tests to test the behavior of the code.

InfiBeyond committed:
Merge branch 'branch-A-JUnit'

* branch-A-JUnit:
  [A-JUnit] ↳ Add JUnit tests

InfiBeyond committed:
[A-Jar]
↳ Package the App as a JAR file

Package the app as an executable JAR file so that it can be distributed easily.

InfiBeyond committed:
[A-JavaDoc]
Add JavaDoc comments to the code.

InfiBeyond committed:
[A-CodingStandard]
Tweak the code to comply with a given coding standard.

InfiBeyond committed:
[Level 9. Find]
Give users a way to find a task by searching for a keyword.

InfiBeyond committed:
Merge branch 'branch-A-JavaDoc'

InfiBeyond committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  [A-CodingStandard] Tweak the code to comply with a given coding standard.

InfiBeyond committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  [Level 9. Find] Give users a way to find a task by searching for a keyword.

InfiBeyond committed:
[A-CheckStyle]

Use checkStyle to detect coding style violations.

InfiBeyond committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  [A-CheckStyle]

InfiBeyond committed:
[Level 10. GUI]
Initialize a GUI to the chatbot.

InfiBeyond committed:
[Level 10. GUI]
Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

InfiBeyond committed:
Add MainWindow.java

InfiBeyond committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add MainWindow.java
  [Level 10. GUI] Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.
  [Level 10. GUI] Initialize a GUI to the chatbot.

InfiBeyond committed:
Generate a new JAR file using Gradle

This bundles JavaFX with the JAR file. This makes sure that the application will work in computers that don't have JavaFX installed.

Generating a new JAR file improves the user inclusivity.

InfiBeyond committed:
[A-Assertions]

Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

InfiBeyond committed:
[A-CheckStyle]

Use checkStyle to detect coding style violations. Update file to ix styling issue and improve user readability.

InfiBeyond committed:
Merge pull request #2 from InfiBeyond/branch-A-Assertions

[A-Assertions]

InfiBeyond committed:
Merge branch 'master' of https://github.com/InfiBeyond/ip

* 'master' of https://github.com/InfiBeyond/ip:
  [A-Assertions]

InfiBeyond committed:
Merge pull request #3 from InfiBeyond/branch-A-CodeQuality

[A-CheckStyle] Improve Code Quality

InfiBeyond committed:
Merge branch 'master' of https://github.com/InfiBeyond/ip

* 'master' of https://github.com/InfiBeyond/ip:
  [A-CheckStyle]

InfiBeyond committed:
[C-DetectDuplicates]

Add the ability to recognize and deal with duplicate items.
Warn users when attempting to add a new task with the same task description with existing task.

InfiBeyond committed:
Merge branch 'branch-C-DetectDuplicates'

* branch-C-DetectDuplicates:
  [C-DetectDuplicates]

InfiBeyond committed:
Fix GUI for display text by changing height and width.

InfiBeyond committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Fix GUI for display text by changing height and width.

# Conflicts:
#	src/main/java/duke/tasks/Task.java

InfiBeyond committed:
Add Ui.png

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Merge branch 'master' of https://github.com/InfiBeyond/ip

* 'master' of https://github.com/InfiBeyond/ip:
  Update README.md

InfiBeyond committed:
Update GUI
Fix height of dialog box to custom user input and bot reply length.

InfiBeyond committed:
Update GUI
Add bot greeting on initialization and disable input after exit command.

InfiBeyond committed:
Fix missing Ui.png
Add updated UI screenshot to repo

InfiBeyond committed:
Update UI.png location

InfiBeyond committed:
Fix styling errors

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Update README.md

InfiBeyond committed:
Improve JUnit Testing

More wholesome testing for Parser and Todo

InfiBeyond committed:
Update Jar file

InfiBeyond committed:
Update credits for code Reuse

Remove .class files

InfiBeyond committed:
Update credits for code reuse
